src/utils/scopes.js

```diff
@@ -155,9 +155,13 @@ export function getScopes(
         ? getSourceBindingVariables(bindings, sourceBindings, key)
         : getBindingVariables(bindings, key);
 
-      // show exception, return, and this variables in innermost scope
-      if (scope.actor === pausedScopeActor) {
-        vars = vars.concat(getSpecialVariables(pauseInfo, key));
+      // On the innermost scope of a frame that is just about to be popped, show
+      // the return value or the exception being thrown as special variables.
+      if (
+        scope.actor === selectedScope.actor &&
+        selectedFrame.id == pauseInfo.frame.id
```

Comment:
we should use `===` because this could hold true for `selectedFrame.id = undefined; pauseInfo.frame.id = 0`